File: benchmarks/runtime/class_field.js
class Vec2 {
    constructor() {
        this.x = 0;
        this.y = 0;
    }
}

function main() {
    let n = 500000;
    let i = 0;
    let sum = 0;
    while (i < n) {
        const v = new Vec2();
        v.x = i;
        v.y = i + 1;
        sum += v.x + v.y;
        i += 1;
    }
    if (sum <= 0) process.exit(1);
    console.log(sum);
}

main();

File: benchmarks/runtime/class_field.lua
local n = 500000
local i = 0
local sum = 0
while i < n do
  local v = { x = 0, y = 0 }
  v.x = i
  v.y = i + 1
  sum = sum + v.x + v.y
  i = i + 1
end
if sum <= 0 then os.exit(1) end
print(sum)
